### PR TITLE
DEV-3390: add subscription settings

### PR DESCRIPTION
**Pull request by nrh-cklimas**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

n/a

#### What's this PR do?

- Adds a route at `/settings/subscription` that shows the user their current plan. This route is only accessibly directly by URL in this PR. Note that the PR doesn't implement the entire page as it appears in Figma. See Jira discussion.
- Converts some related modules to TypeScript and adds tests around them.

#### Why are we doing this? How does it help us?

Part of our Core plan work, and continues the TS/Jest migration.

#### How should this be manually tested? Please include detailed step-by-step instructions.

Happy path, for an org admin:

1. Visit `/settings/subscription/`.
2. Confirm the plan shown and its pricing is accurate.
3. Confirm the "View full pricing comparison" link opens in a new window and goes to our pricing page.
4. Confirm the "Support" link under "Downgrade or Cancel" goes to the support page.
5. Repeat steps 1-4 for orgs with all pricing plans. The Plus plan should show no cost (not free--just omit the pricing).

Route isn't viewable by Hub admins:

1. Visit `/settings/subscription/`.
2. Confirm you get redirected to the pages dashboard.
 
#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

We'd want to document this settings page.

#### Have automated unit tests been added? If not, why?

Yes.

#### How should this change be communicated to end users?

Via documentation.

#### Are there any smells or added technical debt to note?

No.

#### Has this been documented? If so, where?

No.

#### What are the relevant tickets? Add a link to any relevant ones.

https://news-revenue-hub.atlassian.net/browse/DEV-3390

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No. This change isn't linked anywhere yet.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

https://news-revenue-hub.atlassian.net/browse/DEV-3396

### CODE REVIEW

File: spa/src/components/settings/Subscription/SubscriptionPlan.styled.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { EnginePlan } from 'hooks/useContributionPage';
+import styled from 'styled-components';
+
+const planColors = {
+  CORE: '#62ffe3',
+  FREE: '#f2ff59',
+  PLUS: '#f323ff'
+};
```
Comment (GuiMend) [inline spa/src/components/settings/Subscription/SubscriptionPlan.styled.ts:8]: Shouldn't we add these colors to our theme file? We are having to use it in several files already

File: spa/src/constants/orgPlanConstants.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+/**
+ * Annual cost of each plan in USD. These are *only* for display purposes. These
+ * are not used for actual billing. Undefined here means that we don't have a
+ * cost to display.
+ */
+export const PLAN_ANNUAL_COSTS: Record<EnginePlan['name'], number | undefined> = {
+  CORE: 2000,
+  FREE: 0,
+  PLUS: undefined
```
Comment (GuiMend) [inline spa/src/constants/orgPlanConstants.ts:11]: Shouldn't it be 4000 (as in figma) or 12x900 (website price) = 10800 ?

**Review by GuiMend — COMMENTED** [review]
@nrh-cklimas , plus plan doesn't show pricing. Shouldn't it be $4000/year?

Screenshot 2023-04-10 at 18 47 28

File: spa/src/constants/orgPlanConstants.ts (hunk 2 above)
Comment (nrh-cklimas) [reply spa/src/constants/orgPlanConstants.ts:11]: No, we decided to not show this. I'll put a comment in Jira pointing to where this was discussed (far down in the history of the story).

File: spa/src/components/settings/Subscription/SubscriptionPlan.styled.ts (hunk 1 above)
Comment (nrh-cklimas) [reply spa/src/components/settings/Subscription/SubscriptionPlan.styled.ts:8]: Good callout--this was intended to be part of the theme cleanup that I promised in https://github.com/newsrevenuehub/rev-engine/pull/1073#discussion_r1154466382 but got distracted by other stuff in the sprint. I created https://news-revenue-hub.atlassian.net/browse/DEV-3487 to track this and I'll address after this story.

**Review by GuiMend — APPROVED** [review]

**Pull request merged**